detect.go

```diff
@@ -0,0 +1,65 @@
+package main
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/paketo-buildpacks/packit"
+)
+
+type BuildPlanMetadata struct {
+	Launch bool `toml:"launch"`
+}
+
+//go:generate faux --interface Parser --output fakes/parser.go
+type Parser interface {
+	Parse(path string) (hasMri bool, err error)
+}
+
+func Detect(gemfileParser Parser) packit.DetectFunc {
+	return func(context packit.DetectContext) (packit.DetectResult, error) {
+		_, err := os.Stat(filepath.Join(context.WorkingDir, "config.ru"))
```

Comment:
Would it be better to look for the `rack` gem in the Gemfile?

Reply:
I didn't do that because if checked for `rack` gem, detect won't pass for rack-compatible apps. If the user uses a rack-compatible framework like sinatra, then the gemfile does not usually list rack. Looks like rack is then an indirect dependency.

Reply:
Fair enough. Still, it might be able to use a few heuristics to be sure that this needs to detect. Maybe you look first at `Gemfile.lock` for rack because that is the most certain indicator, then maybe check `Gemfile`, and finally check for `*.ru` files on the filesystem.